src/renderers/dom/client/wrappers/ReactDOMSelect.js

```diff
@@ -169,6 +191,7 @@ var ReactDOMSelect = {
     if (__DEV__) {
       checkSelectPropTypes(inst, props);
       warnIfValueIsNull(props);
+      warnIfDuplicateValues(inst);
```

Comment:
Duplicate values is valid for uncontrolled components, so we probably shouldn't warn in that case.